datastore/tests/test_additional_data_geolookups.py

```diff
@@ -0,0 +1,188 @@
+from db.models import Grant
+
+
+class TestAdditionalDataGeoLookup(TestCase):
+    fixtures = ["test_data.json"]
+    EXISTING_AREA = "E01011964"
+
+    def mock_csv_downloads(self, m):
+        # Import Geo lookups data.
+        for areatype, a in GeoLookupSource.SOURCE_URLS.items():
+
+            # mock the CSV downloads
+            url_filename = a["url_lookup"].split("/")[-1]
+            with open(
+                "./datastore/tests/files/geolookups/{}".format(url_filename), "r"
+            ) as infile:
+                m.get("{}".format(a["url_lookup"]), text=infile.read())
+
+            if a.get("url_latlong"):
+                url_filename = a["url_latlong"].split("/")[-1]
+                with open(
+                    "./datastore/tests/files/geolookups/{}".format(url_filename), "r"
+                ) as infile:
+                    m.get("{}".format(a["url_latlong"]), text=infile.read())
+
+    def test_import_geolookups_with_data(self):
+        geo = GeoLookupSource()
+
+        with requests_mock.Mocker() as m:
+            self.mock_csv_downloads(m)
+
+            geo.import_geo_lookups()
+
+            self.assertEqual(len(GeoLookup.objects.all()), 60)
+
+            # check one example
+            geo_object = GeoLookup.objects.filter(areacode=self.EXISTING_AREA)
+            self.assertTrue(len(geo_object), 1)
+            self.assertEqual(geo_object[0].data["areaname"], "Hartlepool 007B")
+            self.assertEqual(
+                round(geo_object[0].data["latitude"], 5), round(54.68510808262816, 5)
+            )
+            self.assertEqual(geo_object[0].data["areatype"], "lsoa")
+            self.assertEqual(geo_object[0].data["rgncd"], "E12000001")
+            # {
+            #     "ewcd": "K04000001",
+            #     "ewnm": "England and Wales",
+            #     "gbcd": "K03000001",
+            #     "gbnm": "Great Britain",
+            #     "ukcd": "K02000001",
+            #     "uknm": "United Kingdom",
+            #     "ladcd": "E06000001",
+            #     "ladnm": "Hartlepool",
+            #     "rgncd": "E12000001",
+            #     "rgnnm": "North East",
+            #     "ctrycd": "E92000001",
+            #     "ctrynm": "England",
+            #     "utlacd": "E06000001",
+            #     "utlanm": "Hartlepool",
+            #     "cauthcd": "E47000006",
+            #     "cauthnm": "Tees Valley",
+            #     "lad17cd": "E06000001",
+            #     "areacode": "E01011964",
+            #     "areaname": "Hartlepool 007B",
+            #     "areatype": "lsoa",
+            #     "msoa11cd": "E02002489",
+            #     "msoa11nm": "Hartlepool 007",
+            #     "ttwa11cd": "E30000215",
+            #     "ttwa11nm": "Hartlepool",
+            #     "msoa11hclnm": "Old Town & Grange",
+            #     "latitude": 54.68510808262816,
+            #     "longitude": -1.223715283044111,
+            # }
```

Comment:
@drkane are these uncommented lines to show the key, value that the test is checking or where left here by mistake?

Reply:
They were there to show me what the test was expecting to see while I was writing the text. Can probably be removed, but could be useful to see what the object is expected to look like?

Reply:
Yes, it can be useful too.